### PR TITLE
Document openshift update process

### PR DESCRIPTION
**Pull request by nshankar13**
Signed-off-by: nshankar13 <nshankar@microsoft.com>

**Description**: Add documentation for updating the OpenShift cluster used in the nightly job. 

Part of: https://github.com/openservicemesh/osm/issues/5003

**Testing done**: Local testing on OpenShift nightly job cluster to perform upgrades. Also results of nightly job, which is now unblocked: https://github.com/openservicemesh/osm/runs/7889995412?check_suite_focus=true

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [X] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [X] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?

### CODE REVIEW

File: docs/development_guide/openshift/openshift_nightly.md [hunk 1]
```diff
@@ -70,3 +70,83 @@ OpenShift 4 requires that any container programming iptables has privileged esca
...
 
 For OSM on OpenShift, the init container must run as privileged. Instructions to enable this feature can be found in the [OSM OpenShift documentation](https://release-v0-9.docs.openservicemesh.io/docs/install/#openshift).
+
+### Upgrading the OpenShift Cluster
+
+As OSM now explicitly lists a kubeVersion in the chart.yaml, the minimum Kubernetes version that OSM is compatible with (and for which Helm will block installations for lower versions), the OpenShift cluster will need to be periodically updated so that the Kubernetes version reflects the kubeVersion of the chart currently in main (for tag `latest-main`). The Kubernetes version that corresponds to each OpenShift Container Platform release can be found on the (release notes)[https://docs.openshift.com/container-platform/4.10/release_notes/ocp-4-10-release-notes.html] on the OpenShift docs site. For instance, to update to Kubernetes version v1.23.x, we will need to update the OpenShift cluster to 4.10.x or above.
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply docs/development_guide/openshift/openshift_nightly.md:76]: Why does the "version reflects the `kubeVersion` of the chart currently in main" branch?  Would it be version for the released version of OSM?

File: docs/development_guide/openshift/openshift_nightly.md [hunk 2]
```diff
@@ -70,3 +70,83 @@ OpenShift 4 requires that any container programming iptables has privileged esca
...
+# Update to the latest version
+oc adm upgrade --to-latest=true
+```
+
+```bash
+# Update to a specific version
+oc adm upgrade --to=<version>
+```
+
+Once you run the upgrade command, you should see the following output as a confirmation that the upgrade is starting:
```
Comment (jsturtevant) [inline docs/development_guide/openshift/openshift_nightly.md:144]: Is the Openshift cluster long lived or do we create it each time we run the e2e suite?

File: docs/development_guide/openshift/openshift_nightly.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nshankar13) [reply docs/development_guide/openshift/openshift_nightly.md:76]: The OpenShift nightly job and OSM NoInstall Nightly jobs both test against the chart currently in main. We currently have another job called "nightly-images" which builds and pushes the chart in main with the tag "latest-main." 

https://github.com/openservicemesh/osm/actions/workflows/nightly-images.yml

File: docs/development_guide/openshift/openshift_nightly.md (hunk 2 above)
Comment (nshankar13) [reply docs/development_guide/openshift/openshift_nightly.md:144]: The cluster is long-lived, we log in to that cluster using OC cli for the nightly job and run the e2e suite.

File: docs/development_guide/openshift/openshift_nightly.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsturtevant) [reply docs/development_guide/openshift/openshift_nightly.md]: ah I missed this was for the nightly-e2e's that makes sense (and I see the title of the doc now 😄 )

**Review by jsturtevant — APPROVED** [review]

**Review by ksubrmnn — APPROVED** [review]
This is so thorough and helpful. Really proud of all the work you've done here! OpenShift is a tricky one :)

**Review by steeling — APPROVED** [review]

**Review by keithmattix — APPROVED** [review]

**Pull request merged**